src/recurrent/matrix/equation.ts

```diff
@@ -157,94 +194,114 @@ class Equation {
    * @param {Matrix} input
    * @returns {Matrix}
    */
-  input(input) {
+  input(input: Matrix): Matrix {
     this.states.push({
       product: input,
-      forwardFn: (product) => {
+      forwardFn: (product: Matrix) => {
+        if (!this.inputValue) return;
+
         product.weights = input.weights = this.inputValue;
       },
+      backpropagationFn: () => {},
     });
+
     return input;
   }
 
   /**
    * connects a matrix via a row
-   * @param {Matrix} m
+   * @param {Matrix} matrix
    * @returns {Matrix}
    */
-  inputMatrixToRow(m) {
+  inputMatrixToRow(matrix: Matrix): Matrix {
     // eslint-disable-next-line @typescript-eslint/no-this-alias
-    const self = this;
-    const product = new Matrix(m.columns, 1);
+    // const self = this;
+    const product = new Matrix(matrix.columns, 1);
+
     this.states.push({
-      left: m,
-      get right() {
-        return self.inputRow;
-      },
       product,
-      forwardFn: rowPluck,
-      backpropagationFn: rowPluckB,
+      left: matrix,
+      // TOFIX: returns `number` but expected `Matrix`
+      // get right() {
```

Comment:
@robertleeplummerjr need help here, please

Reply:
For the time being, to keep it as close to the original as possible, I'd just have an `ts-ignore` there.  The new networks that allow for network composition are replacing these.